Common/Python/PythonInitializer.cs

```diff
@@ -38,22 +40,30 @@ public static class PythonInitializer
         // Used to hold pending path additions before Initialize is called
         private static List<string> _pendingPathAdditions = new List<string>();
 
+        private static string _algorithmLocation;
+
         /// <summary>
-        /// Initialize python
+        /// Initialize python.
+        ///
+        /// See DebuggerHelper.DebugpyThreadInitialization doc for info on why we keep the GIL state
+        /// before calling BeginAllowThreads.
```

Comment:
nit - docs are off now? We can add a nice comment of which case we don't want to call `BeginAllowThreads`